### DISCUSSION TITLE
Subscription Heartbeat / Keep Alive / Ping Pong

### DISCUSSION THREAD

**Discussion by LexanRed** [Netflix/dgs-framework#880] [Q&A]
Hi fellow coders 🧑‍💻

While implementing subscriptions over websockets, I came across some interesting things. 
While working locally, everything seemed working fine. When testing online, websockets where closed after 60 seconds of inactivity. 

We solved it by adding another subscription endpoint which every client subscribes to. It sends a boolean every 55 seconds to prevent the sockets from closing. 

Is this how it is supposed to be? Is there a better solution, like a built in heartbeat / ping pong directly in DGS?

For now it kind of works, it made me curious how you are solving that problem.

**Comment by srinivasankavitha** [comment]
I think we are missing keep-alives in the framework implementation for websockets. So you're right, that should be part of the framework and you should not have to open another endpoint to work around this. Will create a new issue for this and fix in the next release.
